lib/src/layer/tile_layer/tile_layer.dart

```diff
@@ -0,0 +1,549 @@
+import 'dart:async';
+import 'dart:math' as math;
+
+import 'package:collection/collection.dart' show MapEquality;
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_map/flutter_map.dart';
+import 'package:flutter_map/src/core/bounds.dart';
+import 'package:flutter_map/src/core/util.dart' as util;
+import 'package:flutter_map/src/layer/tile_layer/coords.dart';
+import 'package:flutter_map/src/layer/tile_layer/level.dart';
+import 'package:flutter_map/src/layer/tile_layer/tile_manager.dart';
+import 'package:flutter_map/src/layer/tile_layer/tile_transformation.dart';
+import 'package:flutter_map/src/layer/tile_layer/tile_widget.dart';
+import 'package:flutter_map/src/layer/tile_layer/transformation_calculator.dart';
+import 'package:flutter_map/src/map/map.dart';
+import 'package:latlong2/latlong.dart';
+import 'package:tuple/tuple.dart';
+
+part 'tile_layer_options.dart';
+
+class TileLayerWidget extends StatelessWidget {
+  final TileLayerOptions options;
+
+  const TileLayerWidget({Key? key, required this.options}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final mapState = MapState.maybeOf(context)!;
+
+    return TileLayer(
+      mapState: mapState,
+      stream: mapState.onMoved,
+      options: options,
+    );
+  }
+}
+
+class TileLayer extends StatefulWidget {
+  final TileLayerOptions options;
+  final MapState mapState;
+  final Stream<void> stream;
+
+  TileLayer({
+    required this.options,
+    required this.mapState,
+    required this.stream,
+  }) : super(key: options.key);
+
+  @override
+  State<StatefulWidget> createState() {
+    return _TileLayerState();
+  }
```

Comment:
Removing unnecessary statements.

```suggestion
  State<StatefulWidget> createState() => _TileLayerState();
```